CMakeLists.txt

```diff
@@ -10,6 +10,7 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   set(CMAKE_CXX_STANDARD_REQUIRED YES)
 endif()
 
+include(CTest)
```

Comment:
In commit "CTest: Module must be included at the top level" (e7329aed0f711400d7a14ce8d5edbede03b1e75b)

Would it work to move this include down by other includes like `include(GNUInstallDirs)` below around line 45? This compat_find include on line 13 is meant to be a special case, not primary place where includes are located.

Also, in case we do decide to make this include conditional based on whether a build option is set, or whether this is a top-level project it would be better if this happened below where more conditions should be known.